pkg/cluster-runtime/installer.go

```diff
@@ -226,3 +239,40 @@ func (i *Installer) GetCurrentDriver() (registry.Driver, runtime.Driver, error)
 
 	return registryDriver, runtimeDriver, nil
 }
+
+func (i *Installer) setNodeTaints(hosts []net.IP, driver runtime.Driver) error {
+	var (
+		k8snode k8sv1.Node
+		ok      bool
+	)
+	nodeList := k8sv1.NodeList{}
+	if err := driver.List(context.TODO(), &nodeList); err != nil {
+		return fmt.Errorf("failed to list cluster nodes: %v", err)
+	}
+	nodeTaint := make(map[string]k8sv1.Node)
+	for _, node := range nodeList.Items {
+		nodeTaint[getAddress(node.Status.Addresses)] = node
+	}
+
+	for _, ip := range hosts {
+		taints := i.infraDriver.GetHostTaints(ip)
+		if taints != nil {
+			if k8snode, ok = nodeTaint[ip.String()]; ok {
+				k8snode.Spec.Taints = taints
+			}
+			if err := driver.Update(context.TODO(), &k8snode); err != nil {
```

Comment:
bug: should update node if node ip in nodeTaint. Otherwise , no need to do update.